kolena/workflow/model.py

```diff
@@ -218,6 +243,7 @@ def _from_data_with_infer(
         obj.name = data.name
         obj.metadata = data.metadata
         obj.tags = data.tags
+        obj.workflow = cls.workflow
```

Comment:
nit: do we need this?

Reply:
Yes, otherwise equality checks like this fail
```
    def test__init() -> None:
        name = with_test_prefix(f"{__file__}::test__init")
        model = Model(name=name)
        loaded = Model.load(name)
    
>       assert model == loaded
E       AssertionError: assert Model(name='REEK9KB972PG /Users/consoull/workspace/kolena/tests/integration/workflow/test_model.py::test__init', metadata={}, tags=set(), workflow=Workflow(name='Dummy Workflow 🤖', test_sample_type=<class 'tests.integration.workflow.dummy.DummyTestSample'>, ground_truth_type=<class 'tests.integration.workflow.dummy.DummyGroundTruth'>, inference_type=<class 'tests.integration.workflow.dummy.DummyInference'>), infer=None) == Model(name='REEK9KB972PG /Users/consoull/workspace/kolena/tests/integration/workflow/test_model.py::test__init', metadata={}, tags=set(), infer=None)
```

Reply:
Actually I think a better solution is to remove setting the workflow in `_populate_from_other` and also remove this line. Workflow is populated at the class level when getting the Model from `define_workflow`. After that it should not be set.
What do you think?

Reply:
I cannot reproduce where `Model()` and `Model.load()` returns differently. If I add a custom `__eq__` , `assert model == loaded` would pass.

Reply:
The problem appears to be that in `Model.__init__` the [class variable `workflow`](https://github.com/kolenaIO/kolena/blob/trunk/kolena/workflow/model.py#L67) is shadowed by a `workflow` instance variable [here](https://github.com/kolenaIO/kolena/blob/trunk/kolena/workflow/model.py#L206).
When [loading a Model from data we do not set an instance variable for workflow](https://github.com/kolenaIO/kolena/blob/trunk/kolena/workflow/model.py#L209-L223) and instead use the class workflow. This is why
```
name = with_test_prefix(f"{__file__}::test__init")
model = Model(name=name)
loaded = Model.load(name)
assert model.workflow == loaded.workflow # passes
assert model == loaded
*** AssertionError
```
I believe we should update to the solution described in https://github.com/kolenaIO/kolena/pull/315#discussion_r1397679293